hopsworks-api/src/main/java/io/hops/hopsworks/api/user/UserProfileBuilder.java

```diff
@@ -29,77 +31,132 @@
+    u.setBbcGroupCollection(groups);
+    u = userFacade.update(u);
+    usersController.sendConfirmationMail(u);
+    return u;
+  }
+
+  public UserProfileDTO acceptUser(Integer userId, Users newUser) throws UserException, ServiceException {
+    Users u = accept(userId, newUser);
+    return new UserProfileDTO(u);
+  }
+
+  public UserProfileDTO acceptUsers(List<Integer> ids, HttpServletRequest req) {
+    UserProfileDTO userProfileDTO = new UserProfileDTO();
+    Users target = null;
+    for (Integer id : ids) {
+      try {
+        target = accept(id, null);
+        userProfileDTO.addItem(new UserProfileDTO(target));
+      } catch (UserException | ServiceException e) {
+        LOGGER.log(Level.WARNING, "Failed to accept user with id: {0}", id);
       }
-      u.setStatus(UserAccountStatus.ACTIVATED_ACCOUNT);
-      u.setBbcGroupCollection(groups);
-      u = userFacade.update(u);
-      usersController.sendConfirmationMail(u);
+    }
+    if (userProfileDTO.getItems() != null) {
+      userProfileDTO.setCount((long) userProfileDTO.getItems().size());
     } else {
-      throw new UserException(RESTCodes.UserErrorCode.TRANSITION_STATUS_ERROR, Level.WARNING,
-        "status: " + u.getStatus().name() + " to status " + UserAccountStatus.ACTIVATED_ACCOUNT.name());
+      userProfileDTO.setCount(0L);
+    }
+
+    return userProfileDTO;
+  }
+
+  public UserProfileDTO changeRole(Integer userId, String role) throws UserException {
+    Users u = usersController.getUserById(userId);
+    Collection<BbcGroup> groups = new ArrayList<>();
+    BbcGroup bbcGroup = bbcGroupFacade.findByGroupName(role);
+    if (bbcGroup == null) {
+      throw new UserException(RESTCodes.UserErrorCode.ROLE_NOT_FOUND, Level.FINE);
     }
+    groups.add(bbcGroup);
+    u.setBbcGroupCollection(groups);
+    u = userFacade.update(u);
     return new UserProfileDTO(u);
   }
-  
+
+  public Users reject(Integer userId) throws UserException, ServiceException {
+    Users u = usersController.getUserById(userId);
+    u.setStatus(UserAccountStatus.SPAM_ACCOUNT);
+    u = userFacade.update(u);
+    usersController.sendRejectionEmail(u);
+    return u;
+  }
+
   public UserProfileDTO rejectUser(Integer userId) throws UserException, ServiceException {
-    Users u = userFacade.find(userId);
-    if (u != null) {
-      u.setStatus(UserAccountStatus.SPAM_ACCOUNT);
-      u = userFacade.update(u);
-      usersController.sendRejectionEmail(u);
+    Users u = reject(userId);
+    return new UserProfileDTO(u);
+  }
+
+  public UserProfileDTO rejectUsers(List<Integer> ids, HttpServletRequest req) {
+    UserProfileDTO userProfileDTO = new UserProfileDTO();
+    Users target = null;
+    for (Integer id : ids) {
+      try {
+        target = reject(id);
+        userProfileDTO.addItem(new UserProfileDTO(target));
+      } catch (UserException | ServiceException e) {
+        LOGGER.log(Level.WARNING, "Failed to reject user with id: {0}", id);
```

Comment:
sam as above... silently failing on bulk ops... should at least let the user know which ids failed.

Reply:
I added exception if everything failed.